pkg/kn/commands/route/list.go

```diff
@@ -57,13 +57,13 @@ func NewRouteListCommand(p *commands.KnParams) *cobra.Command {
 			case 1:
 				routeList, err = client.ListRoutes(v1alpha12.WithName(args[0]))
 			default:
-				return fmt.Errorf("'kn route list' accepts maximum 1 argument.")
+				return fmt.Errorf("'kn route list' accepts no or one name argument, but not %d arguments.", len(args))
```

Comment:
Is this just me... I find this bit: `accepts no or one name argument` hard to parse. Sorry.

Reply:
maybe "none or a single name argument"   is better ? Hope for some native speaker to kick. @abrennan89 any suggestion for a better wording here ?

Reply:
Perhaps: `'kn route list' does not accept more than one name argument` ?  Since they passed in more than 1.

I don't think we need to show how many they passed in, they can count it if they want :-)

Reply:
> Perhaps: `'kn route list' does not accept more than one name argument` ? Since they passed in more than 1.

That does not explicit mention that no argument is allowed as well. I settled now to a simple message "allows only one additional argument"

Reply:
I like the other changes. Any agreement on this one? I like @duglin's suggestion but would consider something else (if proposed). For me I'd go with the affirmative version: `'kn route list' accepts 0 or 1 argument` or similar